# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCRegMgmt-inl.h
# Commit: 6089b4360edc
# Full Hash: 6089b4360edcfb3a4bda506e37be8ca75a9f15a6
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 6
# Description:
#   Bug 1727084 - Memory64 - Non-atomic-RMW baseline operations. r=yury
#   
#   Implement scalar, SIMD, and atomic loads and stores for the baseline
#   compiler for x64, x86, arm64, and arm32.  Atomic RMW operations and
#   wait/notify are in the next patch.
# ==============================================================================

diff -r b9d32ef0ad83 -r 6089b4360edc js/src/wasm/WasmBCRegMgmt-inl.h
--- a/js/src/wasm/WasmBCRegMgmt-inl.h	Mon Oct 18 07:31:55 2021 +0000
+++ b/js/src/wasm/WasmBCRegMgmt-inl.h	Mon Oct 18 07:31:55 2021 +0000
@@ -212,6 +212,13 @@
 
 RegI32 BaseCompiler::fromI64(RegI64 r) { return RegI32(lowPart(r)); }
 
+RegI32 BaseCompiler::maybeFromI64(RegI64 r) {
+  if (!r.isValid()) {
+    return RegI32::Invalid();
+  }
+  return fromI64(r);
+}
+
 #ifdef JS_PUNBOX64
 RegI64 BaseCompiler::fromI32(RegI32 r) { return RegI64(Register64(r)); }
 #endif